class.mesh-controls.php

```diff
@@ -0,0 +1,455 @@
+					'type'           => 'select',
+					'css_classes'    => array( 'mesh-choose-layout' ),
+					'validation_cb'  => false,
+                    'options_cb'     => array( $this, 'get_template_options' ),
+                    'id'             => 'mesh-sections-template-' . $section->ID,
+				),
+				'display-title' => array(
+					'label'          => __( 'Display Title', 'mesh' ),
+					'type'           => 'checkbox',
+					'css_classes'    => array( 'mesh-section-show-title' ),
+					'show_on_cb'     => false,
+					'validation_cb'  => false,
+				),
+			),
+			'more_options' => array(
+				'css-class' => array(
+					'label'	         => __( 'CSS Class', 'mesh' ),
+					'type'           => 'text',
+					'css_classes'    => array( 'mesh-section-class' ),
+					'show_on_cb'     => false,
+					'validation_cb'  => false,
+				),
+				'collapse' => array(
+					'label'          => __( 'Collapse Padding', 'mesh' ),
+					'type'           => 'checkbox',
+					'css_classes'    => array( 'mesh-section-collapse-input' ),
+					'show_on_cb'     => false,
+					'validation_cb'  => false,
+				),
+				'push-pull' => array(
+					'label'          => __( 'Push Pull', 'mesh' ),
+					'type'           => 'checkbox',
+					'css_classes'    => array( 'mesh-section-push' ),
+					'show_on_cb'     => array( $this, 'show_push_pull' ),
+					'validation_cb'  => false,
+				),
+				'featured_image' => array(
+					'type' => 'media',
+				),
+				'lp-equal' => array(
+					'label'          => __( 'Equalize', 'mesh' ),
+					'type'           => 'checkbox',
+					'css_classes'    => 'mesh-section-equalize',
+					'show_on_cb'     => array( $this, 'show_equalize' ),
+					'validate_cb'    => false,
+				),
+			),
+		);
+
+		$controls = apply_filters( 'mesh_section_controls', $controls );
+
+		if ( $visible ) {
+			$controls = $controls['visible_options'];
+		} else {
+			$controls = $controls['more_options'];
+		}
+
+		print( '<ul class="small-block-grid-1 medium-block-grid-4">' );
+
+		foreach( $controls as $key => $control ) {
+
+			$display_control = true;
+
+			if ( ! empty( $control['show_on_cb'] ) && is_callable( $control['show_on_cb'] ) ) {
+				$display_control = call_user_func_array( $control['show_on_cb'], array( $section, $blocks ) );
+			}
+
+			if ( ! $display_control ) {
+			    continue;
+            }
+
+            if ( ! empty( $control['css_classes'] ) && is_array( $control['css_classes'] ) ) {
+                $css_classes = array_map( 'sanitize_html_class', $control['css_classes'] );
+            } else {
+                $css_classes = array( sanitize_html_class( $control['css_classes'] ) );
+            }
+
+            $css_classes = implode( ' ', $css_classes );
+
+            $underscore_key = str_replace( '-', '_', $key );
+            $current        = get_post_meta( $section->ID, '_mesh_' . esc_attr( $underscore_key ), true );
+            ?>
+            <li class="mesh-section-control-<?php esc_attr_e( $key ); ?>">
```

Comment:
`esc_attr_e()` also runs through the translation filters. Should be replaced with `echo esc_attr()` instead.